dmutils/content_loader.py

```diff
@@ -4,49 +4,49 @@
 import os
 
 
-class ContentLoader(object):
-
-    def __init__(self, manifest, content_directory):
-
-        section_order = self._read_yaml_file(manifest)
+class ContentBuilder(object):
 
+    def __init__(self, manifest, content_directory, yaml_loader):
+        self.yaml_loader = yaml_loader
+        section_order = yaml_loader.read(manifest)
         self._directory = content_directory
-        self._question_cache = {}
-        self.sections = [
+
+        self._all_sections = [
             self._populate_section(s) for s in section_order
         ]
 
-    def get_section(self, requested_section):
+        self.sections = self._all_sections
 
-        for section in self.sections:
+    def get_section(self, requested_section):
+        for section in self._all_sections:
             if section["id"] == requested_section:
                 return section
+        return None
 
     def get_question(self, question):
+        return self.yaml_loader.read(self._directory + question + ".yml")
```

Comment:
This fails when reusing the same `YAMLLoader` instance with two content loaders: `section["questions"]` contains full question dictionaries which leads to a `TypeError: cannot concatenate 'str' and 'dict' objects`